app/celery/tasks.py

```diff
@@ -22,32 +24,52 @@ def send_files_to_dvla(jobs_ids):
-def send_jobs_to_dvla(jobs_ids):
+def send_jobs_to_dvla(job_ids):
     try:
-        failed_jobs = []
-        ensure_local_file_directory()
-        for job_id in jobs_ids:
-            if not get_file_from_s3(current_app.config['DVLA_UPLOAD_BUCKET_NAME'], job_id):
-                failed_jobs.append(job_id)
-
-        dvla_file, successful_jobs, failures = concat_files()
-        failed_jobs += [job_id_from_filename(failed_file) for failed_file in failures]
-
-        if successful_jobs:
-            ftp_client.send_file("{}/{}".format(current_app.config['LOCAL_FILE_STORAGE_PATH'], dvla_file))
+        with LocalDir('job') as job_folder:
+            job_filenames = [get_job_from_s3(job_id) for job_id in job_ids]
+            current_app.logger.info('Sending {} to dvla'.format(job_filenames))
+            dvla_file = concat_files(job_filenames)
 
-        for successful_job in successful_jobs:
-            notify_celery.send_task(
-                name="update-letter-job-to-sent", args=(job_id_from_filename(successful_job),), queue=NOTIFY_QUEUE
-            )
-        for failed_job in failed_jobs:
+            ftp_client.send_file(str(job_folder / dvla_file))
+    except:
+        current_app.logger.exception('FTP app failed to send jobs')
+        task_name = 'update-letter-job-to-error'
+    else:
+        task_name = 'update-letter-job-to-sent'
+    finally:
+        for job_id in job_ids:
             notify_celery.send_task(
-                name="update-letter-job-to-error", args=(failed_job,), queue=NOTIFY_QUEUE
+                name=task_name, args=(job_id,), queue=NOTIFY_QUEUE
             )
-    except FtpException:
-        for job_id in jobs_ids:
-            notify_celery.send_task(
-                name="update-letter-job-to-error", args=(job_id,), queue=NOTIFY_QUEUE
+
+
+@notify_celery.task(name="send-api-notifications-to-dvla")
+@statsd(namespace="tasks")
+def send_api_notifications_to_dvla(filename):
+    # get pipe file from S3
+    with LocalDir('api') as api_folder:
+        get_api_from_s3(filename)
+
+        dvla_file = rename_api_file(filename)
+
+        notification_references = get_notification_references(dvla_file)
+        current_app.logger.info(
+            'Sending {} notifications from {} to dvla'.format(
+                len(notification_references),
+                filename
             )
-    finally:
-        remove_local_file_directory()
+        )
+
+        try:
+            ftp_client.send_file(str(api_folder / dvla_file))
+        except FtpException:
+            # if there's an s3 error we won't know what notifications to update, so only worry about FTP issues
+            current_app.logger.exception('FTP app failed to send api messages')
+            task_name = "update-letter-notifications-to-error"
+        else:
+            task_name = "update-letter-notifications-to-sent"
+
+        notify_celery.send_task(
+            name=task_name, args=(notification_references,), queue=NOTIFY_QUEUE
```

Comment:
How big of a list of notification_references can we send to SQS? Is there a limit?

Reply:
hmm, looking into it SQS suggests a max of 256kb (http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/limits-messages.html) - 10k uuids in SQS format is 712kb 😑 

i'll test locally, and if it's a problem i'll split it into chunks of 1k references

Reply:
yeah i need to chunk it

```
boto.exception.SQSError: SQSError: 400 Bad Request
<?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Error><Type>Sender</Type><Code>InvalidParameterValue</Code><Message>One or more parameters are invalid. Reason: Message must be shorter than 262144 bytes.</Message><Detail/></Error><RequestId>6aa9b883-5bc7-5a0f-8bf7-eac6d15b64e3</RequestId></ErrorResponse>
```